commit 7c61e7a9318a9d506440d276748d2cdc05e07298
Author: Lukas F. Hartmann <lukas@mntre.com>
Date:   Tue Jun 8 03:08:20 2021 +0200

    Revert "swiotlb: don't modify orig_addr in swiotlb_tbl_sync_single"
    
    This reverts commit 16fc3cef33a04632ab6b31758abdd77563a20759.

diff --git a/kernel/dma/swiotlb.c b/kernel/dma/swiotlb.c
index c10e855a03b..a817dc5537b 100644
--- a/kernel/dma/swiotlb.c
+++ b/kernel/dma/swiotlb.c
@@ -726,6 +726,7 @@ void swiotlb_tbl_sync_single(struct device *hwdev, phys_addr_t tlb_addr,
 
 	if (orig_addr == INVALID_PHYS_ADDR)
 		return;
+	orig_addr += (unsigned long)tlb_addr & (IO_TLB_SIZE - 1);
 
 	validate_sync_size_and_truncate(hwdev, orig_size, &size);
 
